File: backend/app/schemas/system.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class ImageEngineStateOut(BaseModel):
    current_engine: str
    engine_label: str
    pending_confirmation_count: int
    pending_listing_ids: list[str]
    last_openai_error: Optional[str]
    last_switch_to_openai_at: Optional[datetime]

File: frontend/src/types/system.ts
export interface ImageEngineState {
  current_engine: "openai" | "gemini"
  engine_label: string
  pending_confirmation_count: number
  pending_listing_ids: string[]
  last_openai_error: string | null
  last_switch_to_openai_at: string | null
}
